### PR TITLE
raftkv: make snapshot consistent

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang PTAL

### CODE REVIEW

File: src/storage/engine/raftkv.rs [hunk 1]
```diff
@@ -103,42 +103,46 @@ impl<T: PdClient, Trans: Transport> RaftKv<T, Trans> {
...
-    pub fn call_command(&self, request: RaftCmdRequest) -> Result<RaftCmdResponse> {
+    pub fn call_command(&self, request: RaftCmdRequest) -> Result<Arc<Event<RaftCmdResponse>>> {
         let finished = Arc::new(Event::new());
         let finished2 = finished.clone();
         let timeout = Duration::from_secs(DEFAULT_TIMEOUT_SECS);
 
         try!(self.router.rl().send_command(request,
                                            box move |resp| {
                                                finished2.set(resp);
+                                               finished2.wait_clear(None);
```
Comment (ngaut) [inline src/storage/engine/raftkv.rs]: Please add comments.

**Comment by ngaut** [comment]
LGTM

**Comment by ngaut** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

**Pull request merged**